storage/txcache/txCache.go

```diff
@@ -70,15 +71,17 @@ func (cache *TxCache) AddTx(tx *WrappedTransaction) (ok bool, added bool) {
 		cache.doEviction()
 	}
 
+	cache.mutAddTx.Lock()
```

Comment:
can you add protection also for remove & clar

Reply:
done